cpp/CMakeLists.txt

```diff
@@ -140,6 +140,25 @@ set(CUDA_nvgraph_LIBRARY ${CUDA_TOOLKIT_ROOT_DIR}/lib64/libnvgraph.so CACHE STRI
 set(TREELITE_DIR ${CMAKE_CURRENT_BINARY_DIR}/treelite CACHE STRING
   "Path to treelite install directory")
 
+set(CUML_INCLUDE_DIR
```

Comment:
Why do we need to include all the subfolders? Why can't it be just a `${PROJECT_SOURCE_DIR}/include`?

Reply:
I agree that it'll be a huge change to every include statement inside `src` folder. But that's far cleaner approach, IMHO.

Reply:
I wanted to try it to see if it’s possible, and to not have to change the include in src if we decided to have a different folder structure during this PR review. Definitely not agains changing the imports before merging the PR if that’s the better way

Reply:
@teju85 Use something like `${cuml_SOURCE_DIR}` is better than `${PROJECT_SOURCE_DIR}`.  Both are generated by CMake, but the later can cause trouble if cuml is used as a git submodule of another library/application.

Reply:
good one! Agreed. Better to use `cuml_SOURCE_DIR`. Thanks @trivialfis

Reply:
I'll update #1134 with such a list of tasks.